test/new-e2e/tests/agent-platform/common/test_client.go

```diff
@@ -281,3 +282,93 @@ func ReadJournalCtl(t *testing.T, client *TestClient, grepPattern string) string
 	}
 	return journalCtlOutput
 }
+
+// DockerTestClient is a helper to run commands on a docker container for tests
+type DockerTestClient struct {
+	host          *components.RemoteHost
+	containerName string
+}
+
+// NewDockerTestClient creates a client to help write tests that run on a docker container
+func NewDockerTestClient(host *components.RemoteHost, containerName string) DockerTestClient {
+	return DockerTestClient{
+		host:          host,
+		containerName: containerName,
+	}
+}
+
+// RunContainer starts the docker container in the background based on the given image reference
+func (c *DockerTestClient) RunContainer(image string) error {
+	// We run an infinite no-op to keep it alive
+	_, err := c.host.Execute(
+		fmt.Sprintf("docker run -d -e DD_HOSTNAME=docker-test --name '%s' '%s' tail -f /dev/null", c.containerName, image),
+	)
+	return err
+}
+
+// Cleanup force-removes the docker container associated to the client
+func (c *DockerTestClient) Cleanup() error {
+	_, err := c.host.Execute(fmt.Sprintf("docker rm -f '%s'", c.containerName))
+	return err
+}
+
+// Execute runs commands on a Docker remote host
+func (c *DockerTestClient) Execute(command string) (output string, err error) {
+	return c.host.Execute(
+		// Run command on container via docker exec,
+		// wrap in a shell call to achieve similar behavior to remote hosts
+		fmt.Sprintf("docker exec %s sh -c '%s'", c.containerName, command),
+	)
+}
+
+// ExecuteWithRetry execute the command with retry
+func (c *DockerTestClient) ExecuteWithRetry(command string) (output string, err error) {
```

Comment:
*Change request*: 
Please add the retry as an input for the function and logs for the error (if those are available somewhere else, please just point it out)

Reply:
Note that this is a near verbatim copy of https://github.com/DataDog/datadog-agent/blob/644be872cf5770eb1b8f6b5cac352c4babafb438/test/new-e2e/tests/agent-platform/common/test_client.go#L178-L192

Avoiding these sorts of copies was what the other PR attempted to do. I can now see a possible way to refactor this to avoid the code duplication, which could make it simpler to add the logs (which is very reasonable).

> add the retry as an input
You mean the number of retries? Having a hardcoded value here doesn't sound terrible as it makes the function easier to use, as long as we don't hit a case where that is important.

Reply:
Refactored into a common function and addressed the logging now.